src/BulletCollision/CollisionDispatch/btCollisionWorld.h

```diff
@@ -263,6 +263,36 @@ class btCollisionWorld
 			return rayResult.m_hitFraction;
 		}
 	};
+    
+    struct ClosestRayResultCallbackWithInfo : public ClosestRayResultCallback
+    {
+        ClosestRayResultCallbackWithInfo(const btVector3& rayFromWorld, const btVector3& rayToWorld)
+        : ClosestRayResultCallback(rayFromWorld, rayToWorld)
+        {
+        }
+        LocalShapeInfo* m_localShapeInfo;
```

Comment:
this m_localShapeInfo has a very short lifetime, so best not to keep it persistent like this, it is intended only during the callback and not any longer, I think.

Reply:
Also, is is ok to keep this class local to the demo?

Reply:
Done.

Reply:
the change in btCollisionWorld.h seems to be still there, can you revert that file?

Reply:
I already did. I'm fixing another issue revealed by mouse picking. Will push soon.